yb-voyager/src/query/sqltransformer/transformer.go

```diff
@@ -154,3 +155,26 @@ func (t *Transformer) MergeConstraints(stmts []*pg_query.RawStmt) ([]*pg_query.R
 
 	return result, nil
 }
+
+func (t *Transformer) RemoveRedundantIndexes(stmts []*pg_query.RawStmt, redundantIndexesMap map[string]string) ([]*pg_query.RawStmt, map[string]*pg_query.RawStmt, error) {
+
+	var sqlStmts []*pg_query.RawStmt
+	removedIndexToStmt := make(map[string]*pg_query.RawStmt) // index object name to raw stmt
+	for _, stmt := range stmts {
+		stmtType := queryparser.GetStatementType(stmt.Stmt.ProtoReflect())
+		if stmtType != queryparser.PG_QUERY_INDEX_STMT {
+			sqlStmts = append(sqlStmts, stmt)
+			continue
+		}
+		objectName := queryparser.GetIndexObjectNameFromIndexStmt(stmt.Stmt.GetIndexStmt())
```

Comment:
As we were discussing, we should try and be more strict about the comparison, not rely on "index ON table" format of the string. 

ideally, use ObjectName (or something similar). Please add a TODO for this.